application/src/main/java/io/redlink/smarti/webservice/ConversationWebservice.java

```diff
@@ -14,264 +14,511 @@
         }
     }
 
-    @ApiOperation(value = "retrieve the results for a template from a specific creator", response = Result.class, responseContainer = "List")
-    @RequestMapping(value = "{id}/template/{template}/{creator}", method = RequestMethod.GET)
-    public ResponseEntity<?> getResults(@PathVariable("id") ObjectId id,
-                                        @PathVariable("template") int templateIdx,
-                                        @PathVariable("creator") String creator) {
-        //TODO: get the Client for the currently authenticated user 
-        final Conversation conversation = conversationService.getConversation(null, id); //TODO: parse the client instead of null
+    @ApiOperation(value = "re-run analysis based on updated tokens/slot-assignments", response = Analysis.class,
+    notes = "<strong>NOT YET IMPLEMENTED!</strong>" + API_ASYNC_NOTE)
+    @RequestMapping(value = "{id}/analysis", method = RequestMethod.POST)
+    public ResponseEntity<Analysis> rerunAnalysis(
+            @PathVariable("id") ObjectId conversationId,
+            @RequestBody Analysis updatedAnalysis,
+            @ApiParam(API_PARAM_CALLBACK) @RequestParam(value = "callback", required = false) URI callback
+    ) {
+        // TODO: check Authentication / clientId
+
+        return ResponseEntity.status(HttpStatus.NOT_IMPLEMENTED).build();
+    }
+
+
+    @ApiOperation(value = "get the extracted tokes in the conversation", response = Token.class, responseContainer = "List")
+    @RequestMapping(value = "{id}/analysis/token", method = RequestMethod.GET)
+    public ResponseEntity<List<Token>> getTokens(
+            @PathVariable("id") ObjectId conversationId
+    ) {
+        // TODO: check Authentication / clientId
+
+        final Conversation conversation = conversationService.getConversation(conversationId);
         if (conversation == null) {
             return ResponseEntity.notFound().build();
-        }
-        //TODO: with authentication in place we need to get the Client from the user and NOT the conversation!!
-        Client client = clientService.get(conversation.getOwner());
-        if(client == null){
-            throw new IllegalStateException("Owner for conversation " + conversation.getId() + " not found!");
+        } else {
+            return ResponseEntity.ok(conversation.getAnalysis().getTokens());
         }
 
-        try {
-            final Template template = conversation.getTemplates().get(templateIdx);
+    }
 
-            return ResponseEntity.ok(conversationService.getInlineResults(client, conversation, template, creator));
-        } catch (IOException e) {
-            return ResponseEntities.serviceUnavailable(e.getMessage(), e);
-        } catch (IndexOutOfBoundsException e) {
+    @ApiOperation(value = "get the (query-)templates in the conversation", response = Template.class, responseContainer = "List")
+    @RequestMapping(value = "{id}/analysis/template", method = RequestMethod.GET)
```

Comment:
Please check with @westei whether we can name the endpoint `intentTemplate` - does not need to be adapter everywhere, but in the URL, it might be nice.